### PR TITLE
Add IsReplayConnection func

### PR DESCRIPTION
**Pull request by m-samiec**
#### Description
GameplayTagContainer used InternalAck as a soft check if we were in a replay connection. We use InternalAck for it's actual purpose, so created a new function - NetConnection::IsReplayConnection().

Engine PR - https://github.com/improbableio/UnrealEngine/pull/214

#### Primary reviewers
@girayozil @improbable-valentyn

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetConnection.h [hunk 1]
```diff
@@ -30,17 +31,19 @@ class SPATIALGDK_API USpatialNetConnection : public UIpConnection
...
 	virtual void UpdateLevelVisibility(const FName& PackageName, bool bIsVisible) override;
 
+	virtual bool IsReplayConnection() const { return false; }
+
 	// These functions don't make a lot of sense in a SpatialOS implementation.
 	virtual FString LowLevelGetRemoteAddress(bool bAppendPort = false) override { return TEXT(""); }
 	virtual FString LowLevelDescribe() override { return TEXT(""); }
 	virtual FString RemoteAddressToString() override { return TEXT(""); }
 	///////
+	//~ End NetConnection Interface
```
Comment (Vatyx) [inline SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetConnection.h]: what's up with these ~

File: SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetConnection.h [hunk 2]
```diff
@@ -21,6 +21,7 @@ class SPATIALGDK_API USpatialNetConnection : public UIpConnection
 public:
 	USpatialNetConnection(const FObjectInitializer& ObjectInitializer);
 
+	//~ Begin NetConnection Interface
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetConnection.h]:
```suggestion
	// Begin NetConnection Interface
```

File: SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetConnection.h (hunk 1 above)
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetConnection.h]:
```suggestion
	// End NetConnection Interface
```

File: SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetConnection.h [hunk 3]
```diff
@@ -30,17 +31,19 @@ class SPATIALGDK_API USpatialNetConnection : public UIpConnection
 	virtual void LowLevelSend(void* Data, int32 CountBits, FOutPacketTraits& Traits) override;
 #endif
 	virtual bool ClientHasInitializedLevelFor(const AActor* TestActor) const override;
-	virtual void Tick() override;
 	virtual int32 IsNetReady(bool Saturate) override;
 
 	/** Called by PlayerController to tell connection about client level visibility change */
 	virtual void UpdateLevelVisibility(const FName& PackageName, bool bIsVisible) override;
 
+	virtual bool IsReplayConnection() const { return false; }
```
Comment (girayozil) [inline SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetConnection.h]: I'd remove `virtual` and add `override`.

Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetConnection.h]:
```suggestion
	virtual bool IsReplayConnection() const override { return false; }
```

**Pull request merged**